packages/player/src/internal/event-tracking/index.ts

```diff
@@ -0,0 +1,65 @@
+import { db } from '../helpers/event-session';
+import { credentialsProviderStore, eventSenderStore } from '../index';
+
+type BaseCommitData = {
+  group: string;
+  name: string;
+  payload: Record<string, unknown>;
+  version: 1 | 2;
+};
+
+/**
+ * Send event to event system scoped to streaming_metrics category.
+ */
+export async function commit(baseCommitData: BaseCommitData) {
+  /*
+    This is a workaround to keep the old nested header structure,
```

Comment:
What's the deal with this? I think Haffy might have added a workaround to the SDK if it's the same problem I'm thinking of?

Reply:
Back-end apparently doesn't use the "enriched header" from event producer yet so I still need these props as well. But if there is a built in work arounds for that now that'd be nice...